eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/charsequence/Trimming.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.charsequence;
+
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+
+/**
+ * Trimming {@link CharSequence} decorator.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class Trimming extends AbstractCharSequence
```

Comment:
`Trimmed` represents a concrete value, i.e. the trimmed version of the decorated `CharSequence`. If we didn't have the rule to omit the base type for decorators the full name would be `TrimmedCharSequence`. It's a special version of the decorated `CharSequence` that's trimmed.

In contrast, `Following` and `UrlRewriting` modify the behavior of the decorated class.

If we had a factory that creates `CharSequences`, `Trimming` would be a decorator to that factory which trims the created `CharSequences`, i.e.
```
CharSequence s = new Trimming(new CharSequenceFactory()).create();
```

Does that make sense?